src/pybind/mgr/dashboard/module.py

```diff
@@ -439,29 +439,30 @@ def serve(self):
 
         class Root(object):
             @cherrypy.expose
-            def index(self):
-                active_uri = module.get_active_uri()
-                if active_uri:
-                    module.log.info("Redirecting to active '%s'", active_uri)
-                    raise cherrypy.HTTPRedirect(active_uri)
-                else:
-                    template = """
-                <html>
-                    <!-- Note: this is only displayed when the standby
-                         does not know an active URI to redirect to, otherwise
-                         a simple redirect is returned instead -->
-                    <head>
-                        <title>Ceph</title>
-                        <meta http-equiv="refresh" content="{delay}">
-                    </head>
-                    <body>
-                        No active ceph-mgr instance is currently running
-                        the dashboard.  A failover may be in progress.
-                        Retrying in {delay} seconds...
-                    </body>
-                </html>
-                    """
-                    return template.format(delay=5)
+            def default(self, *args, **kwargs):
+                if module.get_module_option('standby_behaviour', 'redirect') == 'redirect':
+                    active_uri = module.get_active_uri()
+                    if active_uri:
+                        module.log.info("Redirecting to active '%s'", active_uri)
+                        raise cherrypy.HTTPRedirect(active_uri)
+                    else:
+                        template = """
+                    <html>
+                        <!-- Note: this is only displayed when the standby
+                             does not know an active URI to redirect to, otherwise
+                             a simple redirect is returned instead -->
+                        <head>
+                            <title>Ceph</title>
+                            <meta http-equiv="refresh" content="{delay}">
+                        </head>
+                        <body>
+                            No active ceph-mgr instance is currently running
+                            the dashboard. A failover may be in progress.
+                            Retrying in {delay} seconds...
+                        </body>
+                    </html>
+                        """
+                        return template.format(delay=5)
```

Comment:
@epuertat this causes issues on pylint:
```
************* Module dashboard.module
R:442,12: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

Reply:
@tspmelo any hints on this one ^^ ?

Reply:
Removing the `else` path should fix the issue. I would also suggest to use 
```
    active_uri = module.get_active_uri()
    if active_uri:
        module.log.info("Redirecting to active '%s'", active_uri)
        raise cherrypy.HTTPRedirect(active_uri)
    template = """
    <html>
        <!-- Note: this is only displayed when the standby
                does not know an active URI to redirect to, otherwise
                a simple redirect is returned instead -->
        <head>
            <title>Ceph</title>
            <meta http-equiv="refresh" content="{delay}">
        </head>
        <body>
            No active ceph-mgr instance is currently running
            the dashboard. A failover may be in progress.
            Retrying in {delay} seconds...
        </body>
    </html>
    """
    return template.format(delay=5)
```

Reply:
By removing `else` pylint (even on v1.9.5) is still producing error. I was able to 'reduce' it to warning in case of using 
```
                    if not active_uri:
                        module.log.info("Redirecting to active '%s'", active_uri)
                        raise cherrypy.HTTPRedirect(active_uri)
                        return None
                    template = """
                    <html>
```
and this results in : ```W:448,24: Unreachable code (unreachable)```

Reply:
Better use `pylint: disable=inconsistent-return-statements` to get rid of the lint error.

Reply:
Done